lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Runtime/HostManager/HostManager.h"
+#include "glow/Backends/DeviceManager.h"
+#include "glow/Graph/Context.h"
+
+#include <future>
+
+using namespace glow;
+
+HostManager::HostManager() {
+  provisioner_ = Provisioner();
+  partitioner_ = Partitioner();
+}
+
+HostManager::~HostManager() { clearHost(); }
+unsigned int HostManager::addNetwork(Module *M) {
+  totalCount_++;
+  int networkID = totalCount_;
+  auto dependencyGraph = partitioner_.partition(M);
+  auto executionDAG = provisioner_.provision(dependencyGraph);
+  networks_.emplace(networkID, executionDAG);
```

Comment:
I think we should have a lock around `networks_`? Otherwise onnxifi layer would have to handle synchronization between calls to `initGraph()`, `runGraph()` etc

Reply:
I think it would help it we specify which methods needs to handle concurrency in the host manager.